nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-test/src/test/java/org/nuxeo/directory/test/io/DirectoryEntryJsonWriterTest.java

```diff
@@ -157,4 +158,41 @@ public void testFetched() throws Exception {
         }
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.directory.tests:test-parent-child-directories.xml")
+    public void testParentChildDirectoriesWithSameIds() throws IOException {
```

Comment:
Could you add the expected layout in a comment so assertions are more obvious
Parent 10
└── Sub 20
Parent 20
├── Sub 10
...